wordpress-mobile/WordPress-Android
# 🤖 [Dynamic Cards] Ordering logic

Fixes #19739

## Description
This PR builds the UI models in the right order/position from the API response.

Note: I added a separate issue for the remote feature flag display logic to avoid extending this PR more https://github.com/wordpress-mobile/WordPress-Android/issues/19797

-----

## To Test:
- Apply this patch: [cardmockspatch.patch](https://github.com/wordpress-mobile/WordPress-Android/files/13678298/cardmockspatch.patch)
- Run the app and confirm that you can see two dynamic cards on the top and bottom of the dashboard.
### Alternatively
- Add a break point [here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/CardsBuilder.kt#L37)
- Run and verify that the expected `Dynamic` objects are listed

-----

## Regression Notes

1. Potential unintended areas of impact

    - My site dashboard

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Added unit tests and adapted existing

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
